Realm/RLMApp.h

```diff
@@ -91,6 +96,49 @@ typedef void(^RLMUserCompletionBlock)(RLMSyncUser * _Nullable, NSError * _Nullab
 - (void)loginWithCredential:(RLMAppCredentials *)credentials
           completionHandler:(RLMUserCompletionBlock)completionHandler;
 
+/**
+ Switch to a specified user. If the user is no longer valid e.g Removed then an error will be thrown
+ This call does not invoke any network calls
+ @param syncUser The user you would like to switch to
+ @returns The user you intend to switch to
+ */
+- (RLMSyncUser *)switchUser:(RLMSyncUser *)syncUser;
+
+/**
+ Removes a specified user
+
+ @param syncUser The user you would like to remove
+ @param completionHandler A callback invoked on completion
+*/
+- (void)removeUser:(RLMSyncUser *)syncUser completionHandler:(RLMOptionalErrorBlock)completionHandler;
+
+/**
+ Logs out the current user
+ @param completionHandler A callback invoked on completion
+*/
+- (void)log_out:(RLMOptionalErrorBlock)completionHandler;
+
+/**
+ Logs out a specific user
+ @param syncUser The user to log out
+ @param completionHandler A callback invoked on completion
+*/
+- (void)log_out:(RLMSyncUser *)syncUser completionHandler:(RLMOptionalErrorBlock)completionHandler;
+
+/**
+  A client for the username/password authentication provider which
+  can be used to obtain a credential for logging in,
+  and to perform requests specifically related to the username/password provider.
+*/
+- (RLMUsernamePasswordProviderClient *)usernamePasswordProviderClient;
```

Comment:
Align with other SDKs about naming.  Same below

Reply:
What are you referencing here?

Reply:
There were discussion that the "ProviderClient" thing was a bit too heavy. Not sure what it ended out with. Just ensure you are aligned with other SDKs on the naming is my point.